XouDevSpec/Classes/Model/MangaSearchStruct.swift

```diff
@@ -7,8 +7,54 @@
 
 import Foundation
 
-public class MangaSearchStruct {
-    
-}
+    public struct mangaInfo: Decodable {
+        public var request_hash: String?
+        public var request_cached: Bool?
+        public var request_cache_expiry: Int?
+        public var results = [MangaDetails]()
+        public var last_page : Int?
+
+        public init() {
+            request_hash.self = ""
+            request_cached.self = true
+            request_cache_expiry.self = 0
+            last_page.self = 0
+
+        }
+
+    }
+
+    public struct MangaDetails: Decodable {
+        public var mal_id: Int?
```

Comment:
I noticed that the models have a lot of similar variables. One alternative to avoid duplication, you could make it a class with all the similar variables and use inheritance for the remaining variables. The shift should not be that much work regarding the JSONDecodable.

Reply:
this could also help with generalising the functions for searches and sorting etc...

Reply:
I will attend to doing these fix ups as soon as my remaining objectives are finished = )